### PR TITLE
Add mold linker build

### PR DESCRIPTION
**Pull request by egtvedt**
Allows building the #mold linker, which can then be used in the cross-toolchain by passing the -fuse-ld=mold to the gcc flags. It is much faster than ld or gold.

This requires a C++20 compiler and cmake.

Initially implemented by Arnaud, and HC added configure check for cmake.

Outstanding task to validate compiler is C++20 compatible.

### CODE REVIEW

**Comment by egtvedt** [comment]
Originally implemented in PR #2150, and I adjusted the check for cmake.

Hopefully kept originally author as primary to keep credits.

**Comment by cpackham** [comment]
I've hit the thing to allow the build but as there's no sample that uses this it won't test much (other than checking that the additions don't break anything).

File: config/linker.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+menu "Linkers"
```
Comment (cpackham) [inline config/linker.in:1]: We probably want some indication that there are other linkers. Ideally we'd set things up so bfd, gold and mold are all selectable here. But as the former two are part of binutils it's probably not super easy to do. As a compromise "External Linkers", "Extra Linkers" or "Additional Linkers" might be a better name. Or maybe `comment "BFD enabled in binutils" if BINUTILS_LINKER_LD` and `comment "GOLD enabled in binutils" if BINUTILS_LINKER_GOLD` and

Comment (egtvedt) [reply config/linker.in:1]: I can add these, no problem.

**Comment by egtvedt** [comment]
> I've hit the thing to allow the build but as there's no sample that uses this it won't test much (other than checking that the additions don't break anything).

I ran a local build, and could produce a toolchain with mold available and appears to run as expected.

Can do a bit more testing tomorrow after revisiting your comment.

File: config/linker.in (hunk 1 above)
Comment (egtvedt) [reply config/linker.in:1]: Adjusted to handle enabling both binutils linkers as well.

And syntax that was accepted on my workstation.

**Comment by egtvedt** [comment]
> > I've hit the thing to allow the build but as there's no sample that uses this it won't test much (other than checking that the additions don't break anything).
> 
> I ran a local build, and could produce a toolchain with mold available and appears to run as expected.
> 
> Can do a bit more testing tomorrow after revisiting your comment.

Quick hello world test program executes as expected on target.

Bonus is the executable size reduction from 76kB to 16kB.

**Comment by rawoul** [comment]
Sorry, I haven't had much time to follow the feedback on my PR, and I'm happy to see you are taking things forward ! I've tested toolchains with mold for i686 and aarch64 targets with no issues, building and executing a lot of linux packages, including wpewebkit, qt, etc... I had few issues in the past but with 2.31 I don't have a single problem.

**Pull request merged**